Measurements/Measurements.ipynb

```diff
@@ -0,0 +1,571 @@
+    "\n",
+    "**Input:** A qubit which is guaranteed to be in $|0\\rangle$ or $|1\\rangle$ state.\n",
+    "\n",
+    "**Output:**  `true` if the qubit was in the $|1\\rangle$ state, or `false` if it was in the $|0\\rangle$ state. The state of the qubit at the end of the operation does not matter."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%kata T101_IsQubitOne_Test \n",
+    "\n",
+    "operation IsQubitOne (q : Qubit) : Bool {\n",
+    "    // The operation M will measure a qubit in the Z basis (|0⟩ and |1⟩ basis)\n",
+    "    // and return Zero if the observed state was |0⟩ or One if the state was |1⟩.\n",
+    "    // To answer the question, you need to perform the measurement and check whether the result\n",
+    "    // equals One - either directly or using library function IsResultOne.\n",
+    "    //\n",
+    "    // Type the following: return M(q) == One;\n",
+    "    // Then run the cell using Ctrl/⌘+Enter.\n",
+    "\n",
+    "    // ...\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Task 1.2. Set the qubit to the $|0\\rangle$ state.\n",
+    "\n",
+    "**Input:** A qubit in an arbitrary state.\n",
+    "\n",
+    "**Goal:**  Change the state of the qubit to $|0\\rangle$."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%kata T102_InitializeQubit_Test \n",
+    "\n",
+    "operation InitializeQubit (q : Qubit) : Unit {\n",
+    "    // ...\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Task 1.3. $|+\\rangle$ or $|-\\rangle$?\n",
+    "\n",
+    "**Input:** A qubit which is guaranteed to be in $|+\\rangle$ or $|-\\rangle$ state. As a reminder, $|+\\rangle = \\frac{1}{\\sqrt{2}} \\big(|0\\rangle + |1\\rangle\\big)$, $|-\\rangle = \\frac{1}{\\sqrt{2}} \\big(|0\\rangle - |1\\rangle\\big)$.\n",
+    "\n",
+    "**Output:** `true` if the qubit was in the $|+\\rangle$ state, or `false` if it was in the $|-\\rangle$ state. The state of the qubit at the end of the operation does not matter."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%kata T103_IsQubitPlus_Test \n",
+    "\n",
+    "operation IsQubitPlus (q : Qubit) : Bool {\n",
+    "    // ...\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Task 1.4. $|A\\rangle$ or $|B\\rangle$?\n",
+    "\n",
+    "**Inputs:** \n",
+    "\n",
+    "1. Angle $\\alpha$, in radians, represented as a `Double`.\n",
+    "2. A qubit which is guaranteed to be in $|A\\rangle$ or $|B\\rangle$ state, where $|A\\rangle = \\cos \\alpha |0\\rangle + \\sin \\alpha |1\\rangle$ and $|B\\rangle = - \\sin \\alpha |0\\rangle + \\cos \\alpha |1\\rangle$.\n",
```

Comment:
A qubit which is guaranteed to be in **either the** $|A\\rangle$ or **the** $|B\\rangle$ state